src/main/java/weiliang/bot/UI.java

```diff
@@ -0,0 +1,193 @@
+            if (taskNo > tasks.size() - 1) {
+                return formatMessage("Unable to remove item!");
+            }
+
+            // Mark complete
+            Task task = tasks.get(taskNo);
+            task.complete();
+
+            String message = formatMessage("Nice, I've marked the task as done!");
+            message += "\n" + task;
+            message += "\n" + "Now you have " + tasks.size() + " tasks in the list.";
+            return message;
+        }
+        if (input.startsWith("todo")) {
+            if (input.matches("^todo .+$")) {
+                Task task = new Task(input.replaceFirst("todo ", ""));
+                tasks.add(task);
+                
+                String message = formatMessage("Got it! I've added this task.");
+                message += "\n" + task;
+                message += "\n" + "Now you have " + tasks.size() + " tasks in the list.";
+                return message;
+            } else {
+                throw new DukeException("The description of a todo cannot be empty!");
+            }
+        }
+        if (input.startsWith("deadline")) {
+            if (input.matches("^deadline .+ \\/by .+$")) {
+                String[] inputs = input.replaceFirst("deadline ", "").split(" /by ");
+                Task task = new Deadline(inputs[0], inputs[1]);
+                tasks.add(task);
+                
+                String message = formatMessage("Got it! I've added this task.");
+                message += "\n" + task;
+                message += "\n" + "Now you have " + tasks.size() + " tasks in the list.";
+                return message;
+            } else {
+                throw new DukeException("The description of a deadline cannot be empty!");
+            }
+        }
+        if (input.startsWith("event")) {
+            if (input.matches("^event .+ \\/at .+$")) {
+                String[] inputs = input.replaceFirst("event ", "").split(" /at ");
+                Task task = new Event(inputs[0], inputs[1]);
+                tasks.add(task);
+                
+                String message = formatMessage("Got it! I've added this task.");
+                message += "\n" + task;
+                message += "\n" + "Now you have " + tasks.size() + " tasks in the list.";
+                return message;
+            } else {
+                throw new DukeException("The description of an event cannot be empty!");
+            }
+        }
+        if (input.matches("^find .+$")) {
+            String searchField = input.substring(5);
+            String message = formatMessage("Printing matches!");
+            int i = 0;
+            for (Task task : tasks) {
+                if (task.getDescription().contains(searchField)) {
+                    message += "\n" + (i+1) + "." + task;
+                }
+                i++;
+            }
+            if (i == 0) {
+                message += "\nNo results found!";
+            }
+            return message;
+        }
```

Comment:
I like the clean nature of the messages returned! Perhaps the Ui could do a bit less stuff (e.g. find) and abstract out further? (SLAP)